Add connector permissions to fleet server service account

jedrazb opened the pull request:
## Changes

Add permissions to `fleet-server` service account to enable running [elastic/connectors](https://github.com/elastic/connectors) as an integration. 

Fleet service service account owns and generates api keys that are used by integrations (components) enrolled in fleet. Connectors are (soon) an integrations that can be deployed in agentless to offer Elastic-managed ingestion story. 

There is a corresponding PR in kibana that would grant custom permissions to the `elastic_connectors` package: https://github.com/elastic/kibana/pull/192081

### Explanation of added permission 
- ~`manage_connector` role~ `cluster:admin/xpack/connector/*` - connector service needs to be able to read and write from internal connector indices, there is an ongoing effort to move to this role for connector APIs (see this [PR](https://github.com/elastic/elasticsearch/pull/110128) for reference)
- Index level permission to `.elastic-connectors*` - this is how we currently manage access to internal connector indices, we need read, write and manage access (for refresh operation), this maps to how this is set up in [enterprise-search-server service account](https://github.com/elastic/elasticsearch/blob/be90011e5789a91527ea29124c30d2fa7e97bf50/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/service/ElasticServiceAccounts.java#L68)
- Index level permission to data index
  - Connectors can sync data to indices prefixed with `content-*`, user can decide on the index name. This corresponds to[ search- prefix from enterprise-search-server ](https://github.com/elastic/elasticsearch/blob/be90011e5789a91527ea29124c30d2fa7e97bf50/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/service/ElasticServiceAccounts.java#L52)
  - `.search-acl-filter-*` - this is a required (and hardcoded in few places) prefix for index storing data related to document-level security logic. If connector index name is e.g. `content-google-drive`, its corresponding ACL index will be: `.search-acl-filter-content-google-drive`

in `enterprise-search-server` we defined index-level permissions to be `read,write,manage`, we can be more granual with  `"read", "write","monitor","create_index","auto_configure","maintenance"` (this would at least prevent deletion of the index)

jedrazb committed:
Add connector permissions to fleet server service account

jedrazb committed:
[Security] Add permissions to manage connectors for fleet-server service account

jedrazb committed:
Merge branch 'main' into add-connector-permissions-to-fleet

jedrazb commented:
@elasticmachine merge upstream

elasticmachine committed:
Merge branch 'main' into add-connector-permissions-to-fleet

elasticsearchmachine commented:
Pinging @elastic/es-security (Team:Security)

elasticsearchmachine commented:
Pinging @elastic/ingestion-team (Team:Search - Extract & Transform)

elasticsearchmachine commented:
Pinging @elastic/ent-search-eng (Team:SearchOrg)

jedrazb committed:
Fix tests

jedrazb committed:
Merge branch 'add-connector-permissions-to-fleet' of github.com:jedrazb/elasticsearch into add-connector-permissions-to-fleet

jedrazb committed:
Fix tests

jedrazb committed:
Fix typ again (tm)

seanstory reviewed (approved):
LGTM from the Search team side.

artem-shelkovnikov commented on x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/service/ElasticServiceAccounts.java:
```diff
@@ -81,7 +81,7 @@ final class ElasticServiceAccounts {
         new RoleDescriptor(
             NAMESPACE + "/fleet-server",
-            new String[] { "monitor", "manage_own_api_key", "read_fleet_secrets" },
+            new String[] { "monitor", "manage_own_api_key", "read_fleet_secrets", "manage_connector" },
```
```suggestion
            new String[] { "monitor", "manage_own_api_key", "read_fleet_secrets", "cluster:admin/xpack/connector/*" },
```

artem-shelkovnikov commented on x-pack/plugin/security/qa/service-account/src/javaRestTest/java/org/elasticsearch/xpack/security/authc/service/ServiceAccountIT.java:
```diff
@@ -112,7 +112,8 @@ public class ServiceAccountIT extends ESRestTestCase {
                 "manage_own_api_key",
-                "read_fleet_secrets"
+                "read_fleet_secrets",
+                "manage_connector"
```
```suggestion
                "cluster:admin/xpack/connector/*"
```

artem-shelkovnikov commented on docs/reference/rest-api/security/get-service-accounts.asciidoc:
```diff
@@ -66,7 +66,8 @@ GET /_security/service/elastic/fleet-server
         "manage_own_api_key",
-        "read_fleet_secrets"
+        "read_fleet_secrets",
+        "manage_connector"
```
```suggestion
        "cluster:admin/xpack/connector/*"
```

seanstory committed:
switch to connector/* vs manage_connectors

@jakelandis pointed out that we don't need connector secrets, which is the only difference between these too. We don't have a pretty name for the narrower permissions, but we don't need one here.

Co-authored-by: Artem Shelkovnikov <lavatroublebubble@gmail.com>

seanstory commented:
buildkite test this

seanstory commented:
@elasticmachine merge upstream

elasticmachine committed:
Merge branch 'main' into add-connector-permissions-to-fleet

seanstory commented:
Bump @elastic/fleet , would love to get a cursory review from one of y'all before we merge this.

nchaulet reviewed (approved):
LGTM 🚀
